### PR TITLE
Update cog loading and handling

### PR DESCRIPTION
**Pull request by lgaan**
### Description

This will update cog loading and handling for the next release of the library. Cogs now are registered with an `add_cog` method within `ipc.server.Server`.

### Checklist

- [x] This PR has been tested.

### CODE REVIEW

**Review by ShineyDev — CHANGES_REQUESTED** [review]
This should be made a non-breaking change by deprecating the old method and allowing both.

File: docs/modules/server.rst [hunk 1]
```diff
@@ -23,7 +23,5 @@ It then calls the method linked to said route and returns the payload to the cli
 
 .. currentmodule:: discord.ext.ipc.server
 
-.. autofunction:: route
```
Comment (ShineyDev) [inline docs/modules/server.rst:26]: How do you expect anybody to know that this is deprecated if you remove it from the documentation!

**Review by ShineyDev — CHANGES_REQUESTED** [review]
I'm still not sure I love the design of this feature, but here are some nit-picks until we have time for a discussion about it.

File: docs/modules/server.rst (hunk 1 above)
Comment (lgaan) [reply docs/modules/server.rst:26]: Forgot to add back in after adding back the deprecated methods.

**Review by ShineyDev — CHANGES_REQUESTED** [review]
I'll review this again when you complete the previous review.

**Comment by jgayfer** [comment]
Would love to see this come to fruition 🙂

**Pull request closed without merging**